app/uk/gov/hmrc/cgtpropertydisposalsstubs/controllers/EnrolmentStoreProxyController.scala

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cgtpropertydisposalsstubs.controllers
+
+import akka.actor.ActorSystem
+import com.google.inject.Inject
+import play.api.mvc.{Action, AnyContent, ControllerComponents, Result}
+import uk.gov.hmrc.cgtpropertydisposalsstubs.models.EnrolmentRequest
+import uk.gov.hmrc.cgtpropertydisposalsstubs.util.Logging
+import uk.gov.hmrc.play.bootstrap.controller.BackendController
+
+import scala.concurrent.ExecutionContext
+
+class EnrolmentStoreProxyController @Inject()(
+  cc: ControllerComponents,
+  system: ActorSystem
+)(implicit ec: ExecutionContext)
+    extends BackendController(cc)
+    with Logging {
+
+  def allocateEnrolmentToGroup(groupId: String, enrolmentKey: String): Action[AnyContent] = Action { implicit request =>
+    request.body.asJson.fold[Result] {
+      logger.warn("Could not find enrolment action payload")
+      BadRequest
+    } { enrolmentAction =>
+      enrolmentAction
+        .validate[EnrolmentRequest]
+        .fold(
+          { error =>
+            logger.warn(s"Invalid JSON payload: [payload : $error]")
+            BadRequest
+          }, { enrolmentAction =>
+            (groupId, enrolmentKey) match {
+              case ("ABCEDEFGI123400", "HMRC-CGT-PD~1234567800") => Created
```

Comment:
the cgt reference number satisfied the regex `^X[A-Z]CGTP[0-9]{9}$` - the bit after `HMRC-CGT-PD~` needs to be updated to reflect that

Reply:
also, we won't have control over the cgt reference id in general unless we code for them specifically in the `SubscriptionProfiles`. We'd have control over the group id though since we can modify this in the auth login wizard